### PR TITLE
Add 'magpie config' command for managing CLI configuration

### LINKED ISSUE #14: CLI: Missing 'config' command documented in user guide
## Problem

The user guide documents a `magpie config` command:

```bash
magpie config --server https://artifacts.swccdc.com --token mgp_xxx...
```

But this command doesn't exist:

```
Error: No such command 'config'.
```

## Current State

Configuration works via:
1. `--server` / `--token` options on each command
2. `MAGPIE_SERVER` / `MAGPIE_TOKEN` environment variables
3. Manual creation of `~/.magpie/config.toml`

## Suggested Fix

Either:
1. **Add the `config` command** to write/update the config file (matches docs)
2. **Update the user guide** to document the actual configuration methods

Option 1 is preferred for better UX.

## Proposed `config` command

```bash
# Set server and token
magpie config --server http://localhost:8080 --token mgp_...

# Show current config
magpie config --show

# Clear config
magpie config --clear
```

---

*Issue created with AI assistance (Claude Code w/ Opus 4.5) during acceptance testing.*

### PR DESCRIPTION
**Pull request by duplico**
## Summary

- Adds the `magpie config` command that was documented in the user guide but not implemented
- Supports `--server URL` and `--token TOKEN` to set persistent configuration values in `~/.magpie/config.toml`
- Supports `--show` to display current configuration (with masked tokens for security)
- Supports `--clear` to remove the configuration file

## Test plan

- [x] All 420 unit tests pass
- [x] New command tests cover all options (set, show, clear, validation)
- [x] Tests verify TOML output is valid and handles special characters
- [x] Linting passes with ruff check and format

Closes #14

Generated with Claude Code (claude-opus-4-5-20251101)

### CODE REVIEW

File: src/magpie/cli/commands/config_cmd.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    Args:
+        config_path: Path to the config file.
+        server: Server URL to write (or None to keep existing).
+        token: Token to write (or None to keep existing).
+    """
+    # Load existing config to preserve values not being set
+    existing = cli_config.load_config(config_path)
+    final_server = server if server is not None else existing.client.server
+    final_token = token if token is not None else existing.client.token
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/magpie/cli/commands/config_cmd.py:46]: Fixed in 70f6fd6. Added documentation in the `_write_config` docstring that explains this limitation:

```python
Note:
    Empty string values are treated the same as the existing value, not as
    a way to clear individual settings. Use --clear to remove all config.
```

This approach keeps the implementation simple while making the behavior clear. The `--clear` flag is the intended way to remove all configuration.

File: tests/unit/test_cli_config_cmd.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+@pytest.fixture
+def config_dir(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> Path:
+    """Create a temporary config directory and patch the default path."""
+    config_path = tmp_path / ".magpie" / "config.toml"
+    monkeypatch.setattr(
+        "magpie.cli.config.DEFAULT_CONFIG_PATH",
+        config_path,
+    )
+    return config_path
```
[the remark this answers is not in the document]
Comment (duplico) [reply tests/unit/test_cli_config_cmd.py]: Fixed in 70f6fd6. Renamed the fixture from `config_dir` to `config_path` since it returns the path to the config file, not a directory:

```python
@pytest.fixture
def config_path(tmp_path: Path, monkeypatch: pytest.MonkeyPatch) -> Path:
    """Create a temporary config path and patch the default path."""
    path = tmp_path / ".magpie" / "config.toml"
    ...
```

Also updated all usages throughout the test file.

File: src/magpie/cli/commands/config_cmd.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    # Write TOML manually (tomllib is read-only)
+    lines = ["[client]"]
+    if final_server:
+        # Escape backslashes and quotes in TOML strings
+        escaped_server = final_server.replace("\\", "\\\\").replace('"', '\\"')
+        lines.append(f'server = "{escaped_server}"')
+    if final_token:
+        escaped_token = final_token.replace("\\", "\\\\").replace('"', '\\"')
+        lines.append(f'token = "{escaped_token}"')
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/magpie/cli/commands/config_cmd.py]: Fixed in 70f6fd6. Added `tomli_w` as a dependency and now use it for proper TOML serialization instead of manual string escaping:

```python
import tomli_w

# Build config dict for tomli_w
client_config: dict[str, str] = {}
if final_server:
    client_config["server"] = final_server
if final_token:
    client_config["token"] = final_token

config_data = {"client": client_config}
config_path.write_bytes(tomli_w.dumps(config_data).encode())
```

This properly handles all TOML escaping including control characters, Unicode, and all the edge cases. The existing tests for special characters and backslashes continue to pass.

File: src/magpie/cli/commands/config_cmd.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        return
+
+    # Set values
+    _write_config(config_path, server, token)
+
+    # Show what was set
+    if server:
+        click.echo(f"Server set to: {server}")
+    if token:
+        click.echo(f"Token set to: {'*' * 8}{token[-4:] if len(token) > 4 else '****'}")
```
[the remark this answers is not in the document]
Comment (duplico) [reply src/magpie/cli/commands/config_cmd.py]: Fixed in 70f6fd6. Created a dedicated `_mask_token()` function with a constant `TOKEN_MASK = "********"` to standardize the masking behavior:

```python
# Standard mask for hiding sensitive token values
TOKEN_MASK = "********"


def _mask_token(token: str) -> str:
    """Mask a token for display, showing only last 4 characters."""
    if len(token) > 4:
        return TOKEN_MASK + token[-4:]
    return TOKEN_MASK
```

Now both the set and show operations use this same function, so the masking is consistent everywhere.

**Comment by duplico** [comment]
Added `# nosec B105` comment to suppress the bandit false positive on line 13.

The `TOKEN_MASK = "********"` constant is a display placeholder used to mask sensitive token values in the CLI output -- it is not an actual password or secret. Bandit's B105 check incorrectly flags this as a hardcoded password.

Verified:
- `bandit -r src/` now passes with no issues
- All unit tests pass (427 passed)
- Ruff lint and format checks pass

---
*Generated with Claude Code (Opus 4.5)*
